vkcs/kubernetes/resource_cluster.go

```diff
@@ -272,23 +291,72 @@ func ResourceKubernetesCluster() *schema.Resource {
 				Computed:    true,
 				Description: "Enables syncing of security policies of cluster. Default value is false.",
 			},
+			"cluster_type": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          clusterTypeStandard,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{clusterTypeStandard, clusterTypeRegional}, false)),
+				Description:      "Type of the kubernetes cluster, may be `standard` or `regional`. Default type is `standard`.",
+			},
 		},
-		Description: "Provides a kubernetes cluster resource. This can be used to create, modify and delete kubernetes clusters.",
+		CustomizeDiff: resourceKubernetesClusterCustomizeDiff,
+		Description:   "Provides a kubernetes cluster resource. This can be used to create, modify and delete kubernetes clusters.",
 	}
 }
 
-func resourceKubernetesClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	config := meta.(clients.Config)
-	containerInfraClient, err := config.ContainerInfraV1Client(util.GetRegion(d, config))
-	if err != nil {
-		return diag.Errorf("error creating container infra client: %s", err)
+func resourceKubernetesClusterCustomizeDiff(_ context.Context, diff *schema.ResourceDiff, _ any) error {
+	newClusterType := diff.Get("cluster_type").(string)
+	switch newClusterType {
+	case clusterTypeStandard:
+		if masterCount, ok := diff.GetOk("master_count"); ok {
+			mCount := masterCount.(int)
+			if mCount != 1 && mCount != 3 && mCount != 5 {
+				return fmt.Errorf("master_count for standard cluster type if set must be one of 1, 3, 5")
+			}
+		}
+
+		if _, ok := diff.GetOk("availability_zone"); !ok {
+			return fmt.Errorf("availability_zone must be specified for cluster with the type standard")
+		}
+
+		if !diff.GetRawConfig().GetAttr("availability_zones").IsNull() {
+			return fmt.Errorf("do not specify availability_zones for cluster with the type standard")
+		}
+	case clusterTypeRegional:
+		if masterCount, ok := diff.GetOk("master_count"); ok {
+			mCount := masterCount.(int)
+			if mCount != 3 && mCount != 5 {
+				return fmt.Errorf("master_count for regional cluster type if set must be equal 3 or 5")
+			}
+		}
+
+		if !diff.GetRawConfig().GetAttr("availability_zone").IsNull() {
```

Comment:
It seems that you can acknowledge this without an use of raw config, by `GetOk`.

Reply:
It's not working in this place, we need to check new users config. When we use GetOk we getting cloud values

Reply:
If so, why would you use it 5 lines above? First of all, you should avoid using raw config, because diff is processed in several steps, and you should work with value from the last one, not the first. Also, it's considered experimental, as doc says, so it's better to avoid using it when not necessary.